dandi/validate.py

```diff
@@ -44,12 +90,101 @@ def validate_bids(
         report_path = report_path.format(
             log_dir=log_dir,
         )
+
     validation_result = validate_bids_(
         paths,
         schema_version=schema_version,
         report_path=report_path,
     )
-    return dict(validation_result)
+    our_validation_result = []
+    origin = ValidationOrigin(
+        name="bidsschematools",
+        version=bidsschematools.__version__,
+    )
+
+    for path in validation_result["path_tracking"]:
+        # Hard-coding exclusion here pending feature + release in:
+        # https://github.com/bids-standard/bids-specification/issues/1272
+        if path.endswith((".ERRORS", ".ERRORS.json")):
+            continue
+        dataset_path = _get_set_path(path, "dataset_description.json")
+        dandiset_path = _get_set_path(path, "dandiset.yaml")
```

Comment:
note that doing that for **every** path could be avoidably expensive and the same folder cannot belong to different datasets or dandisets.  Get path's folder (if not a folder), and cache results.  but see above to re-use existing `find_parent_directory_containing`

Reply:
354a99255cc527e48606a1ad41f7ce4a8bb2cbed